### ISSUE TITLE
Upgrade React package to 16.3.1

### ISSUE THREAD

**Issue by jeremymy-zl** [Microsoft/fast-dna#236]
### Expected behavior 
Expected to have the latest React version.

### Actual behavior
Latest version is 16.3.1, however version used is <16.
16.3.1 has some breaking changes, @K05-MSFT and we thought it is better to upgrade now to avoid further upgrade work.

**Comment by K05-MSFT** [comment]
@nicholasrice @chrisdholt, just a heads up, the 16.3.x also marked some life-cycle APIs as UNSAFE_, that would be nice to get addressed now.

UNSAFE_componentWillMount()
UNSAFE_componentWillReceiveProps()
UNSAFE_componentWillUpdate()

**Comment by chrisdholt** [comment]
@K05-MSFT - Yep, we've looked pretty closely at the now unsafe lifecycle events. As we continue to move components into our new architecture, we'll be watching to ensure that these are not migrated into the new repo.

Currently, I believe there are a couple instances of these methods on some packages' local test environments (fast-viewer-react). In terms of code that is built for consumption, the only place that these lifecycle events exist are in the JSS-Manager. We have an existing task #175 to update the JSS manager to use the new "context-api" which was also released. I've also created a specific issue to track the updates to the safe API #258 which will likely occur during the same refactor for v16.3.x
